src/bsoncxx/json.cpp

```diff
@@ -43,12 +43,28 @@ void bson_free_deleter(std::uint8_t* ptr) {
 
 }  // namespace
 
-std::string BSONCXX_CALL to_json(document::view view) {
+enum class jsonMode { LEGACY, CANONICAL, RELAXED };
+
+std::string _to_json_helper(document::view view, jsonMode mode)
```

Comment:
There is no need to use leading _ in C++. Just move it up into the unnamed namespace, which accomplishes the same thing better. You are flirting with undefined behavior here for no good reason.

Reply:
@acmorrow generally I agree. Out of morbid curiosity could you elaborate on the comment about undefined behavior? Are you just referring to potential conflicts when mangling?

Reply:
https://stackoverflow.com/a/228797

Basically what is here is legal (I think?) because _to_json isn't at global scope, it is at namespace scope, but that is too close for comfort.

- __to_json would be definitely illegal
- _To_json would be definitely illegal.

Why get that close to the line when just putting it in the unnamed namespace achieves the same effect. Leading underscores in any context are ill-advised. Best to avoid them.

Reply:
Correct.